NomNomAPI/Services/FoodItemService/FoodItemService.cs

```diff
@@ -53,6 +66,34 @@ public async Task<IEnumerable<FoodItem>> GetAllFoods()
             return foods;
         }
 
+        public async Task<FoodItem?> GetSingleFood(int id)
+        {
+            var food = await _context.foodItems.FindAsync(id);
+            if (food == null)
+                return null;
```

Comment:
It would make more sense to throw an error here since it is thrown in controllers level anyway. That way you would avoid same if checks and isolate logic better, so it would have higher cohesion and more inline with single responsibility principle